src/main/java/pl/lukakan/showstracker/user/UserService.java

```diff
@@ -0,0 +1,136 @@
+package pl.lukakan.showstracker.user;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.lukakan.showstracker.user.model.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class UserService {
+    private final UserRepository userRepository;
+    private final UserRoleRepository userRoleRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserService(UserRepository userRepository, UserRoleRepository userRoleRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.userRoleRepository = userRoleRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public void registerUser(UserDto userDto) {
+        String rawPassword = userDto.getPassword();
+        String encryptedPassword = passwordEncoder.encode(rawPassword);
+        userDto.setPassword(encryptedPassword);
+        User user = fromDto(userDto);
+        userRepository.save(user);
+    }
+
+    private User fromDto(UserDto userDto) {
+        User user = new User();
+        user.setUserName(userDto.getUserName());
+        user.setEmail(userDto.getEmail());
+        user.setPassword(userDto.getPassword());
+        user.setFirstName(userDto.getFirstName());
+        user.setLastName(userDto.getLastName());
+        Set<UserRole> defaultUserRole = Set.of(new UserRole(user, Role.ROLE_USER));
+        user.setUserRoles(new HashSet<>(defaultUserRole));
```

Comment:
Nie możemy od razu user.setUserRoles(Set.of(new UserRole(user, Role.ROLE_USER))); ?

Reply:
Możemy. Już poprawiam :)